### PR TITLE
adds tag search to CDAP header

### PR DESCRIPTION
**Pull request by astewes**
*  Adds search icon/input to CDAP header
*  Accounts for styling of .navbar-form containing the search input
*  Updates $state.go so that tag searches go to the tag filter view
*  Adds styling for the tag filtered results view (removes search input, displays total number of results, and styling for individual tag objects)
*  Removes "f" and "p" hotkeys from main.js
*  Removes pin search references
*  Accounts for responsiveness of CDAP header

Header with no errors:
search

Header with errors:
searchwitherrors

### CODE REVIEW

File: cdap-ui/app/directives/search/search.html [hunk 1]
```diff
@@ -15,32 +15,18 @@
...
-          <a role="menuitem" class="clearfix">
-            <span class="pull-left">Pin</span>
-            <span class="fa fa-thumb-tack pull-right"></span>
-          </a>
-        </li>
-      </ul>
+           type="text"
+           class="form-control"
+           ng-class="{'search-active': MySearchCtrl.displaySearchBox}"
+           placeholder="search tags"
```
Comment (ajainarayanan) [inline cdap-ui/app/directives/search/search.html]: Probably should be ```Search Tags```

File: cdap-ui/app/features/search/templates/search-objects-with-tags.html [hunk 2]
```diff
@@ -15,95 +15,101 @@
...
-
-      <div ng-repeat="tObject in SearchController.filteredTObjects = (SearchObjectWithTagsController.taggedObjects | filter: SearchObjectWithTagsController.searchTxt ) ">
-        <div ng-if="tObject.type === 'application'">
-          <a class="btn btn-default"
-              ui-sref="apps.detail.overview.status({namespace: tObject.namespaceId, appId: tObject.appId})"
-          >
-            <span class="{{tObject.icon}}"></span>
+  <div class="col-xs-12">
+    <p class="results-count" ng-if="SearchController.filteredTObjects.length">
+      <span>{{SearchController.filteredTObjects.length}}</span>
```
Comment (ajainarayanan) [inline cdap-ui/app/features/search/templates/search-objects-with-tags.html]: Is it possible for us to use ```ngPluralize``` here (https://docs.angularjs.org/api/ng/directive/ngPluralize)?

**Comment by ajainarayanan** [comment]
TODO in this PR:
1. We should remove the main search page where we display all available tags in the system. Right now this is a costly operation and we will be showing duplicate entries if we tag two things with the same tag.
2. The search and notification icons are changing (to make more room for when we both). 
3. This might be changed based on the design changes for (2) but the search icon looks a little out of place. We should probably re-visit them.

**Comment by nitinmotgi** [comment]
There is a jitter when closing the search box.

**Comment by astewes** [comment]
@nitinmotgi jitter you're referring to should be resolved in my latest commit.

**Comment by elfenheart** [comment]
:+1: LGTM

**Pull request merged**